cfe_internal/enterprise/federation/federation.cf

```diff
@@ -300,29 +300,34 @@ bundle agent transport_user
           expression => not(returnszero("$(default:paths.semanage) fcontext -l | grep '$(home)/.ssh(/.*)?'", "useshell")),
           if => fileexists("$(home)");
     enabled.selinux_enabled::
+        # For all the files below it must be true that if they exist they need
```

Comment:
I appreciate your explanation here but have to say this block has become rather unreadable. I wonder if we could do better? The logic reads to me as correct after some time looking at it.

Reply:
Well, I know how to make this a lot nicer in Python. Or any language with functions and loops, really. But no clue how to make it nicer in CFEngine policy. Maybe @nickanderson can help polish this later.

Reply:
We could use a list of files and then create an array with those files as keys and the results of the SELinux labels as values and then search for a false value in `getindices()` on the array. But that's similarly ugly, AFAICT.